### PR TITLE
More consistent IGSIO hardware module

### PR DESCRIPTION
**Pull request by drouin-simon**
* Added a log window to the config window to know what is going on.
* Improved server launching.

### CODE REVIEW

File: IbisHardwareIGSIO/ibishardwareIGSIOsettingswidget.cpp [hunk 1]
```diff
@@ -45,6 +54,9 @@ void IbisHardwareIGSIOSettingsWidget::UpdateUI()
     }
 
     ui->autoStartLastConfigCheckBox->setChecked( m_igsio->GetAutoStartLastConfig() );
+
+    // Display HW module log in the textEdit box
+    //ui->logTextEdit
```
Comment (nist-anka) [inline IbisHardwareIGSIO/ibishardwareIGSIOsettingswidget.cpp]: Do you intend to enable this commented code?

**Comment by nist-anka** [comment]
I have  one comment in the code. Nothing serious. I built and ran it to see it.

File: IbisHardwareIGSIO/ibishardwareIGSIOsettingswidget.cpp (hunk 1 above)
Comment (drouin-simon) [reply IbisHardwareIGSIO/ibishardwareIGSIOsettingswidget.cpp]: I removed the comment

File: IbisHardwareIGSIO/plusserverinterface.cpp [hunk 2]
```diff
@@ -62,27 +68,37 @@ bool PlusServerInterface::StartServer(const QString& configFilePath)
...
+    QThread * thread = QThread::currentThread();
+    QTime timeWaited;
+    timeWaited.start();
+    while( timeWaited.elapsed() < 20000 && m_state == Starting )
+    {
+        thread->msleep( 100 );
+        QApplication::processEvents();
+    }
+
+    return GetState() == Running;
```
Comment (hgueziri) [inline IbisHardwareIGSIO/plusserverinterface.cpp]: In the case of a timeout, should we set the server m_state to Timeout (or something similar) before returning?

Comment (drouin-simon) [reply IbisHardwareIGSIO/plusserverinterface.cpp]: Good question. I had this in the previous version but removed it. Now I'm thinking we should probably just kill the process if we have a timeout. We can report it in the log window.

Comment (hgueziri) [reply IbisHardwareIGSIO/plusserverinterface.cpp]: Indeed, killing the process sounds cleaner.

**Comment by drouin-simon** [comment]
> Not directly related to the modifications, but when the server fails to launch, the error window pops up [(here)](https://github.com/IbisNeuronav/Ibis/blob/a6c738b1ffe677970f591f732cae10ecb0b1fe06/IbisHardwareIGSIO/ibishardwareIGSIO.cpp#L407) behind the setting window and is modal to Ibis main window. The setting window and the main window are frozen, which makes it hard to close the popup warning window.

Yes. I'm thinking we should not pop up this dialog at all. The log window should have the connection details and maybe we could simply have a connection label on the main window to indicate whether all the servers are connected and running. You can open the window if you want more details

**Comment by nist-anka** [comment]
I agrre about not popping up the dialog, log is detailed enough.

**Pull request merged**